profiler/upload.go

```diff
@@ -69,10 +70,18 @@ func (p *profiler) doRequest(bat batch) error {
 	if err != nil {
 		return err
 	}
+	ctx := context.Background()
+	if p.cfg.uploadTimeout != 0 {
```

Comment:
I know why we have this check, without it it's effectively `WithDeadline(time.Now())`.  But I do think we should have a reasonable default.  If 10s isn't appropriate, we can lower it but I don't think blocking indefinitely is what we want.

I think we might also want a test for this in `upload_test.go` with a small `WithUploadTimeout`

Reply:
also, might change this check to be `if p.cfg.uploadTimeout <= 0`

Reply:
Sorry, approved, I somehow missed the `DefaultUploadTimeout` var.  This is a better approach.  I would still change the defensive check on upload to `<=`.

Reply:
3db6059a8c1769f36f8d2a361ff7e8ffa3d60281 - what do you think?